grammar/Expr.g4

```diff
@@ -17,9 +17,20 @@ grammar Expr;
 /* To match the whole input */
 fullexpr: expr EOF;
 
-shift: TIME (op=('+' | '-') expr)?;
+shift: TIME shift_expr?;
+
+shift_expr
```

Comment:
As identified by @tbittar it seems there is still some expressions that cannot be parsed correctly, like `x[t - 2*d + 1].

To fix it, it might be enough to move `shiftMuldiv` higher than `shiftAddsub` : the order matters here to respect math operators precedence !